experiments/AMIP/moist_mpi_earth/coupler_driver.jl

```diff
@@ -141,88 +151,116 @@ coupler_sim = CouplerSimulation(
 include("./push_pull.jl")
 
 # init conservation info collector
-atmos_pull!(coupler_sim)
-atmos_push!(coupler_sim)
-land_pull!(coupler_sim)
+atmos_pull!(cs)
+atmos_push!(cs)
+land_pull!(cs)
 reinit!(atmos_sim.integrator)
 reinit!(land_sim.integrator)
 
-mode_name == "amip" ? (ice_pull!(coupler_sim), reinit!(ice_sim.integrator)) :
-(ocean_pull!(coupler_sim), reinit!(ocean_sim.integrator))
+mode_name == "amip" ? (ice_pull!(cs), reinit!(ice_sim.integrator)) : nothing
+mode_name == "aquaplanet" ? (ocean_pull!(cs), reinit!(ocean_sim.integrator)) : nothing
 
 if !is_distributed && energy_check && mode_name == "aquaplanet"
     conservation_check = OnlineConservationCheck([], [], [], [], [], [])
-    check_conservation(conservation_check, coupler_sim, atmos_sim, land_sim, ocean_sim, ice_sim) # TODO: check this still works for aquaplanet after AMIP
+    check_conservation(conservation_check, cs, atmos_sim, land_sim, ocean_sim, ice_sim)
 end
 
 # coupling loop
-@show "Starting coupling loop"
-
-# step in time
-walltime = @elapsed for t in ((tspan[1] + Δt_cpl):Δt_cpl:tspan[end])
-    cs = coupler_sim
-
-    date = current_date(t)
-
-    @calendar_callback :(@show(date), date1 += Dates.Month(1)) date date1
-
-    ## Atmos
-    atmos_pull!(cs)
-    step!(atmos_sim.integrator, t - atmos_sim.integrator.t, true) # NOTE: instead of Δt_cpl, to avoid accumulating roundoff error
-
-    atmos_push!(cs)
-
-    ## Slab land
-    land_pull!(cs)
-    step!(land_sim.integrator, t - land_sim.integrator.t, true)
-
-    ## Slab ocean
-    if cs.mode.name == "aquaplanet"
-        ocean_pull!(cs)
-        step!(ocean_sim.integrator, t - ocean_sim.integrator.t, true)
+function solve_coupler!(cs, energy_check)
+    @show "Starting coupling loop"
+
+    atmos_sim = cs.model_sims.atm
+    land_sim = cs.model_sims.lnd
+    ocean_sim = cs.model_sims.ocn
+    ice_sim = cs.model_sims.ice
+    Δt_cpl = cs.Δt
+    tspan = cs.tspan
```

Comment:
Maybe something like? Would need to do some renaming
```suggestion
@unpack model_sims,  Δt, tspan = cs
@unpack atm, lnd, ocn = model_sims
```